### PR TITLE
[No issue] Simplify study session filters

### PR DESCRIPTION
**Pull request by her0e1c1**
## Related issue

None

## Summary

- Simplify Study Start tag filtering with explicit Any/All matching and Clear-only actions.
- Keep selected tags visible first, show eight unselected tags initially, and progressively disclose the rest.
- Display unrestricted score boundaries as a dash while preserving accessible no-limit descriptions.
- Preserve keyboard focus when selected or newly revealed tags move in the collapsed list.
- Bound large persisted tag selections in a vertical scroll region.

## Testing

- mise run check (116 test files, 641 tests)
- Storybook tests for StudySessionTagFilter, StudySessionFilters, and StudySessionStart (21 tests)
- Manual Storybook QA at desktop and 320px widths

### CODE REVIEW

File: src/features/deck-filter/ui/ScoreRange.tsx [hunk 1]
```diff
@@ -77,14 +78,15 @@ const ScoreSelect = ({
...
-        { label: "No limit", value: NO_LIMIT_VALUE },
+        { label: NO_LIMIT_LABEL, value: NO_LIMIT_VALUE },
         ...props.scores.map((score) => ({ label: displayScore(score), value: String(score) })),
       ]}
       onChange={(event) =>
         props.onChange(event.currentTarget.value === NO_LIMIT_VALUE ? null : Number(event.currentTarget.value))
       }
     />
     <p id={`${props.id}-description`} className="sr-only">
+      {props.value == null ? `No ${props.label.toLowerCase()} score. ` : ""}
```
Comment (her0e1c1) [inline src/features/deck-filter/ui/ScoreRange.tsx]: **[P1] Keep the no-limit meaning available even when a boundary is set.** This description explains the dash only while `props.value == null`. When Minimum or Maximum already has a numeric value, the select still offers `-`, but assistive technology only gets the generic “Include cards…” description, so users cannot tell that choosing `-` removes that boundary. Please describe the dash unconditionally (for example, `A dash means no minimum score. …`) and add a test for a non-null current value.

File: src/pages/study-session-start/ui/StudySessionTagFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        )}
+      </div>
+
+      {hiddenTagCount > 0 ? (
+        <button
+          type="button"
+          aria-expanded={expanded}
+          aria-controls={tagListId}
+          className="min-h-touch rounded-control px-1 text-caption font-semibold text-accent-primary underline-offset-4 hover:underline focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-focus"
+          onClick={() => setExpanded((current) => !current)}
```
Comment (her0e1c1) [inline src/pages/study-session-start/ui/StudySessionTagFilter.tsx]: **[P2] Keep newly revealed tags in the forward tab sequence.** The disclosure button comes after the tag list. When a keyboard user activates it, the newly rendered checkboxes are inserted before the still-focused button, so the next Tab skips all newly revealed tags. Please either place the disclosure before the controlled content or move focus to the first newly revealed tag after expansion, and cover the keyboard interaction in a test.

File: src/pages/study-session-start/ui/StudySessionTagFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            All
+          </label>
+        </div>
+      </fieldset>
+
+      <div id={tagListId}>
+        {visibleTags.length === 0 ? (
+          <p className="text-caption text-ink-muted">No tags available.</p>
+        ) : (
+          <TagList>
```
Comment (her0e1c1) [inline src/pages/study-session-start/ui/StudySessionTagFilter.tsx]: **[P2] Bound the selected-tag list as well.** The collapsed limit applies only to unselected tags. Existing persisted selections can contain every tag because the previous UI supported “All”, so `visibleTags` can still render 100+ chips; this new `TagList` also no longer enables its max-height scrolling behavior. Please pass something like `hasManyItems={visibleTags.length > 30}` or otherwise cap the selected section, and add a story/test for a large all-selected state.

**Review by her0e1c1 — COMMENTED** [review]
One accessibility regression and two keyboard/large-state UX issues need attention before merge. The page-first placement, selected-first ordering, stale-tag recovery, and focused test coverage are otherwise solid. The P1 should be fixed before merge; the two P2 findings should also be handled to keep the simplified UI usable across keyboard and legacy persisted states.

File: src/features/deck-filter/ui/ScoreRange.tsx (hunk 1 above)
Comment (her0e1c1) [reply src/features/deck-filter/ui/ScoreRange.tsx]: Addressed in 21b21f44. The accessible description now explains the dash unconditionally for both minimum and maximum selects, and the unit test covers numeric current values.

File: src/pages/study-session-start/ui/StudySessionTagFilter.tsx (hunk 3 above)
Comment (her0e1c1) [reply src/pages/study-session-start/ui/StudySessionTagFilter.tsx]: Addressed in 21b21f44. Tag choices now use a bounded vertical scroll region when more than 30 chips are visible, with a 120-selected unit test and Storybook state.

File: src/pages/study-session-start/ui/StudySessionTagFilter.tsx (hunk 2 above)
Comment (her0e1c1) [reply src/pages/study-session-start/ui/StudySessionTagFilter.tsx]: Addressed in 21b21f44. Keyboard activation of Show more now moves focus to the first newly revealed tag; the test verifies forward Tab order and focus preservation when collapsing.
